projects/refer-and-monitor-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/contact/ContactRepository.kt

```diff
@@ -101,6 +102,26 @@ interface ContactRepository : JpaRepository<Contact, Long> {
         endTime: String,
         previousExternalReference: String?
     ): Int
+
+    @Query(
+        """
+            select 
+                nsi.soft_deleted,
+                nsi.active_flag, 
+                last_updated_by.distinguished_name
+            from contact
```

Comment:
Yes that's the idea - to provide some extra information when the contact no longer exists, so that we can categorise the errors.  You're right though the original query only handled the scenario where the contact had been soft-deleted, I've updated it to handle hard-deleted contacts as a result of NSI termination.